### ISSUE TITLE
StopIteration: index 4 is out of bounds for axis 1 with size 2

### ISSUE THREAD

**Issue by hszkf** [crild/facies_net#2]
Is there any reason why I got the error? I think the error is because of keras updates but Im not sure how to fix it.. 
![screenshot from 2018-09-28 10-33-43](https://user-images.githubusercontent.com/42924203/46214915-22754480-c30a-11e8-8a57-d6672d0e85c5.png)

**Comment by hszkf** [comment]
Could you please list the requirements details for this? Thanks!

**Comment by ice-pice** [comment]
@hasiftwistcode : were you able to debug this issue?
I am facing the same error. Using tensorflow 1.5.0. Thanks.

**Comment by ice-pice** [comment]
I figured it out. 
The issue is with number of classes. When I changed the number of classes to 9, instead of 2, it worked.
Although the code has more following errors.
